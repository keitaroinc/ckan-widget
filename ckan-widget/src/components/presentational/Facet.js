import React, { Component } from 'react';


class Facet extends Component {
  render() {
    const facets = this.props.facetsInfo;
    const title = this.props.title;
    const infoList = [];

    for (const f in facets) {
      infoList.push(
        <li className="list-group-item d-flex justify-content-between align-items-center" key={f} onClick={() => this.props.onClick(`${this.props.facetKey}:${facets[f].name}`)}>
          {facets[f].display_name}
          {' '}
          <span className="badge badge-primary">{facets[f].count}</span>
        </li>,
      );
    }
    return (
      <div className="card my-3">
        <div className="card-header bg-secondary text-white">
            <h5 className="card-title mb-auto">{title}</h5>
        </div>
          <ul className="list-group list-group-flush">
            {infoList}
          </ul>
      </div>
    );
  }
}

export default Facet;
