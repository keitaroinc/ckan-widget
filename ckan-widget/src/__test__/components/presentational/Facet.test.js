import React from 'react';
import { shallow } from 'enzyme';
import Facet from '../../../components/presentational/Facet';


let component;
let onClick;

describe('Facet', () => {
  beforeEach(() => {
    onClick = jest.fn()
    component = shallow(
      <Facet
        onClick={onClick}
        facetKey={'key'}
        title={'Facet title'}
        facetsInfo={[
          {count: 1, display_name: 'Facet1', name: 'facet1'},
          {count: 3, display_name: 'Facet2', name: 'facet2'},
          {count: 4, display_name: 'Facet3', name: 'facet3'},
          {count: 5, display_name: 'Facet4', name: 'facet4'},
          {count: 5, display_name: 'Facet5', name: 'facet5'},
          {count: 4, display_name: 'Facet6', name: 'facet6'},
          {count: 12, display_name: 'Facet7', name: 'facet7'},
          {count: 17, display_name: 'Facet8', name: 'facet8'},
          {count: 18, display_name: 'Facet9', name: 'facet9'},
          {count: 19, display_name: 'Facet10', name: 'facet10'},
        ]} />
    );
  });

  it('should be defined', () => {
    expect(Facet).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should receive props', () => {
    expect(component.instance().props.facetKey).toBe('key')
    expect(component.instance().props.title).toBe('Facet title')
    expect(component.instance().props.facetsInfo[0].count).toBe(1)
    expect(component.instance().props.facetsInfo[0].display_name).toBe('Facet1')
    expect(component.instance().props.facetsInfo[0].name).toBe('facet1')
  });

  it('should handle facet click event', () => {
    const li = component.find('li.list-group-item').first()
    li.simulate('click')
    expect(onClick.mock.calls.length).toEqual(1);
  })

  it('should expand facet list', () => {
    expect(component.state().collapsed).toBe(true)
    const a = component.find('a.btn').first()
    a.simulate('click')
    expect(component.state().collapsed).toBe(false)
  })
});