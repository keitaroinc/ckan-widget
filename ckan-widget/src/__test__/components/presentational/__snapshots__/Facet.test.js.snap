// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facet should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Facet
    facetKey="key"
    facetsInfo={
        Array [
            Object {
              "count": 1,
              "display_name": "Licence Ouverte Version 2.0",
              "name": "license-3",
            },
          ]
    }
    onClick={[Function]}
    title="Facet title"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-header bg-secondary text-white"
>
          <h5
                    className="card-title mb-auto"
          >
                    Facet title
          </h5>
</div>,
        <ul
          className="list-group list-group-flush"
>
          <li
                    className="list-group-item d-flex justify-content-between align-items-center"
                    onClick={[Function]}
          >
                    Licence Ouverte Version 2.0
                     
                    <span
                              className="badge badge-primary"
                    >
                              1
                    </span>
          </li>
</ul>,
      ],
      "className": "card my-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h5
            className="card-title mb-auto"
>
            Facet title
</h5>,
          "className": "card-header bg-secondary text-white",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Facet title",
            "className": "card-title mb-auto",
          },
          "ref": null,
          "rendered": "Facet title",
          "type": "h5",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="list-group-item d-flex justify-content-between align-items-center"
              onClick={[Function]}
>
              Licence Ouverte Version 2.0
               
              <span
                            className="badge badge-primary"
              >
                            1
              </span>
</li>,
          ],
          "className": "list-group list-group-flush",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Licence Ouverte Version 2.0",
                " ",
                <span
                  className="badge badge-primary"
>
                  1
</span>,
              ],
              "className": "list-group-item d-flex justify-content-between align-items-center",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              "Licence Ouverte Version 2.0",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                  "className": "badge badge-primary",
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header bg-secondary text-white"
>
            <h5
                        className="card-title mb-auto"
            >
                        Facet title
            </h5>
</div>,
          <ul
            className="list-group list-group-flush"
>
            <li
                        className="list-group-item d-flex justify-content-between align-items-center"
                        onClick={[Function]}
            >
                        Licence Ouverte Version 2.0
                         
                        <span
                                    className="badge badge-primary"
                        >
                                    1
                        </span>
            </li>
</ul>,
        ],
        "className": "card my-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h5
              className="card-title mb-auto"
>
              Facet title
</h5>,
            "className": "card-header bg-secondary text-white",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Facet title",
              "className": "card-title mb-auto",
            },
            "ref": null,
            "rendered": "Facet title",
            "type": "h5",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="list-group-item d-flex justify-content-between align-items-center"
                onClick={[Function]}
>
                Licence Ouverte Version 2.0
                 
                <span
                                className="badge badge-primary"
                >
                                1
                </span>
</li>,
            ],
            "className": "list-group list-group-flush",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Licence Ouverte Version 2.0",
                  " ",
                  <span
                    className="badge badge-primary"
>
                    1
</span>,
                ],
                "className": "list-group-item d-flex justify-content-between align-items-center",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                "Licence Ouverte Version 2.0",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 1,
                    "className": "badge badge-primary",
                  },
                  "ref": null,
                  "rendered": 1,
                  "type": "span",
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
